packages/neo4j-driver/src/session-rx.js

```diff
@@ -104,6 +107,45 @@ export default class RxSession {
     return this._runTransaction(ACCESS_MODE_WRITE, work, transactionConfig)
   }
 
+
+  /**
+   * Executes the provided unit of work in a {@link READ} reactive transaction which is created with the provided
+   * transaction configuration.
+   * @public
+   * @param {function(txc: RxManagedTransaction): Observable} work - A unit of work to be executed.
+   * @param {TransactionConfig} transactionConfig - Configuration for the enclosing transaction created by the driver.
+   * @returns {Observable} - A reactive stream returned by the unit of work.
+   */
+   executeRead (work, transactionConfig) {
+    return this._executeInTransaction(ACCESS_MODE_READ, work, transactionConfig)
+  }
+
+  /**
+   * Executes the provided unit of work in a {@link WRITE} reactive transaction which is created with the provided
+   * transaction configuration.
+   * @public
+   * @param {function(txc: RxManagedTransaction): Observable} work - A unit of work to be executed.
+   * @param {TransactionConfig} transactionConfig - Configuration for the enclosing transaction created by the driver.
+   * @returns {Observable} - A reactive stream returned by the unit of work.
+   */
+  executeWrite (work, transactionConfig) {
+    return this._executeInTransaction(ACCESS_MODE_WRITE, work, transactionConfig)
+  }
+
+  /**
+   * @private
+   * @param {function(txc: RxManagedTransaction): Observable} work
+   * @param {TransactionConfig} transactionConfig
+   * @returns {Observable}
+   */
+  _executeInTransaction (accessMode, work, transactionConfig) {
+    const wrapper = txc => new RxManagedTransaction({
```

Comment:
wouldn't it be easier if `RxManagedTransaction` constructor accepted `txc` as its argument instead of an arbitrary object?

Reply:
The bind will only change places at the end. I don't want to keep reference from the transaction inside the ManagedTransaction to avoid the client code by pass the encapsulation and commit/rollback/close it anyway.

Reply:
what about:
```typescript
class RxManagedTransaction {

constructor(txc) {
    this._run = txc.run.bind(txc)
    this._isOpen = tcx.isOpen.bind(txc)
  }
```
so you don't have to repeat the destructuring every time you call the constructor

Reply:
I've moved to the binding to a static factory method called `fromTransaction`. It makes easier to pass the function as parameter.